### PR TITLE
feat: secret sharing supports expiry on view count

### PR DESCRIPTION
**Pull request by ShubhamPalriwala**
# Description 📣
Secret Sharing Feature has the following changes:
- Got rid of Name (Key Value pairs are removed)
- Multi line input to share anything across securely
- Expire secrets either based on time or on number of views
- Docs open in a new tab
- Updated docs as per new options including new org level placement
- Public shared page shows left time/views based on the secret
- And some Prettier linting

## Type ✨

- [x] Bug fix
- [ ] New feature
- [ ] Breaking change
- [x] Documentation

# Tests 🛠️

```sh
# Here's some code block to paste some code snippets
```

---

- [x] I have read the [contributing guide](https://infisical.com/docs/contributing/getting-started/overview), agreed and acknowledged the [code of conduct](https://infisical.com/docs/contributing/getting-started/code-of-conduct). 📝

### CODE REVIEW

File: backend/src/services/secret-sharing/secret-sharing-service.ts [hunk 1]
```diff
@@ -43,9 +54,18 @@ export const secretSharingServiceFactory = ({
...
     }
+    if (sharedSecret.expiresAfterViews != null && sharedSecret.expiresAfterViews >= 0) {
+      if (sharedSecret.expiresAfterViews === 0) {
+        await secretSharingDAL.deleteById(sharedSecretId);
+        return;
+      }
+      await secretSharingDAL.updateById(sharedSecretId, {
+        expiresAfterViews: sharedSecret.expiresAfterViews - 1
+      });
+    }
```
Comment (sheensantoscapadngan) [inline backend/src/services/secret-sharing/secret-sharing-service.ts:68]: use $decr

File: docs/documentation/platform/secret-sharing.mdx [hunk 2]
```diff
@@ -1,37 +1,42 @@
 ---
 title: "Secret Sharing"
 sidebarTitle: "Secret Sharing"
-description: "Learn how to share time-bound secrets securely with anyone on the internet."
+description: "Learn how to share time or view-count bound secrets securely with anyone on the internet."
```
Comment (sheensantoscapadngan) [inline docs/documentation/platform/secret-sharing.mdx:4]: I think users should be able to use both restrictions?

File: frontend/src/views/ShareSecretPublicPage/ShareSecretPublicPage.tsx [hunk 3]
```diff
@@ -46,18 +50,26 @@ export const ShareSecretPublicPage = () => {
...
+              (timeDifference % millisecondsPerDay) / millisecondsPerHour
+            );
+            const minutesRemaining = Math.floor(
+              (timeDifference % millisecondsPerHour) / millisecondsPerMinute
+            );
+            const secondsRemaining = Math.floor((timeDifference % millisecondsPerMinute) / 1000);
+            setTimeLeft(`${hoursRemaining}h ${minutesRemaining}m ${secondsRemaining}s`);
+          }
+        } else if (data.expiresAfterViews) {
+          setTimeLeft(`${data.expiresAfterViews - 1} more views`);
```
Comment (sheensantoscapadngan) [inline frontend/src/views/ShareSecretPublicPage/ShareSecretPublicPage.tsx:72]: hmm.. I think we don't need to show this prompt to the public user? it's better that we limit the stuff they know about the shared secret

File: frontend/src/views/ShareSecretPublicPage/ShareSecretPublicPage.tsx [hunk 4]
```diff
@@ -46,18 +50,26 @@ export const ShareSecretPublicPage = () => {
...
+            setTimeLeft("Expired");
+          } else {
+            const hoursRemaining = Math.floor(
+              (timeDifference % millisecondsPerDay) / millisecondsPerHour
+            );
+            const minutesRemaining = Math.floor(
+              (timeDifference % millisecondsPerHour) / millisecondsPerMinute
+            );
+            const secondsRemaining = Math.floor((timeDifference % millisecondsPerMinute) / 1000);
+            setTimeLeft(`${hoursRemaining}h ${minutesRemaining}m ${secondsRemaining}s`);
```
Comment (sheensantoscapadngan) [inline frontend/src/views/ShareSecretPublicPage/ShareSecretPublicPage.tsx:69]: here as well

**Review by sheensantoscapadngan — COMMENTED** [review]
did we already configure a cleanup job to delete expired shared secrets? or is that something that we don't want to do?

File: backend/src/services/secret-sharing/secret-sharing-service.ts (hunk 1 above)
Comment (ShubhamPalriwala) [reply backend/src/services/secret-sharing/secret-sharing-service.ts:68]: thanks, missed this.

File: docs/documentation/platform/secret-sharing.mdx (hunk 2 above)
Comment (ShubhamPalriwala) [reply docs/documentation/platform/secret-sharing.mdx:4]: I found it weird to have to setup both hence felt the UX might be better off to just let them choose the option and set only that.

File: frontend/src/views/ShareSecretPublicPage/ShareSecretPublicPage.tsx (hunk 3 above)
Comment (ShubhamPalriwala) [reply frontend/src/views/ShareSecretPublicPage/ShareSecretPublicPage.tsx:72]: aye nice idea! done for both 💪🏼

**Comment by ShubhamPalriwala** [comment]
> did we already configure a cleanup job to delete expired shared secrets? or is that something that we don't want to do?

Yepz, we have a cleanup cronjob for time based expiries! And for view based, it happens on the last view

File: docs/documentation/platform/secret-sharing.mdx (hunk 2 above)
Comment (sheensantoscapadngan) [reply docs/documentation/platform/secret-sharing.mdx:4]: For reference, bitwarden does it like the following

**Review by sheensantoscapadngan — APPROVED** [review]

**Review by maidul98 — APPROVED** [review]

**Pull request merged**